lib/task/migrate/migrations/arMigration0177.class.php

```diff
@@ -0,0 +1,245 @@
+        'onDelete' => 'ON DELETE CASCADE',
+      ),
+      array(
+        'tableName' => 'information_object',
+        'columnName' => 'display_standard_id',
+        'refTableName' => 'term',
+        'newConstraintName' => 'information_object_FK_8',
+        'onDelete' => 'ON DELETE SET NULL',
+      ),
+    );
+
+    foreach ($foreignKeys as $foreignKey)
+    {
+      // Get actual contraint name
+      $sql = 'SELECT CONSTRAINT_NAME FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE ';
+      $sql .= 'WHERE TABLE_NAME=:table_name AND COLUMN_NAME=:column_name ';
+      $sql .= 'AND REFERENCED_TABLE_NAME=:ref_table_name;';
+      $oldConstraintName = QubitPdo::fetchColumn($sql, array(
+        ':table_name' => $foreignKey['tableName'],
+        ':column_name' => $foreignKey['columnName'],
+        ':ref_table_name' => $foreignKey['refTableName'],
+      ));
+
+      // Stop if the foreign key is missing
+      if (!$oldConstraintName)
+      {
+        throw new Exception(sprintf(
+          "Could not find foreign key for '%s' column on '%s' table.",
+          $foreignKey['columnName'],
+          $foreignKey['tableName']
+        ));
+      }
+
+      // Having the same name requires to drop and add in two statements
+      $sql = 'ALTER TABLE %s DROP FOREIGN KEY %s;';
+      QubitPdo::modify(sprintf(
+        $sql,
+        $foreignKey['tableName'],
+        $oldConstraintName
+      ));
+
+      $sql = 'ALTER TABLE %s ADD CONSTRAINT %s FOREIGN KEY (%s) ';
+      $sql .= 'REFERENCES %s (id) %s;';
+      QubitPdo::modify(sprintf(
+        $sql,
+        $foreignKey['tableName'],
+        $foreignKey['newConstraintName'],
+        $foreignKey['columnName'],
+        $foreignKey['refTableName'],
+        $foreignKey['onDelete']
+      ));
+    }
+
+    // Restore NOT NULL constraint on culture columns (removed on migration 172)
+    $i18nTables = array(
+      'accession_i18n',
+      'acl_group_i18n',
+      'actor_i18n',
+      'contact_information_i18n',
+      'deaccession_i18n',
+      'event_i18n',
+      'function_object_i18n',
+      'information_object_i18n',
+      'menu_i18n',
+      'note_i18n',
+      'other_name_i18n',
+      'physical_object_i18n',
+      'property_i18n',
+      'relation_i18n',
+      'repository_i18n',
+      'rights_i18n',
+      'setting_i18n',
+      'static_page_i18n',
+      'taxonomy_i18n',
+      'term_i18n'
+    );
+    $defaultCulture = sfConfig::get('sf_default_culture', 'en');
+
+    foreach($i18nTables as $i18nTable)
+    {
+      $baseTable = str_replace('_i18n', '', $i18nTable);
+
+      // Update possible NULL values added in between migrations.
+      // The culture and id together are a unique key on the i18n tables so
+      // there shouldn't be more than a NULL value per object id.
+      $sql = 'UPDATE %s SET culture=:cul WHERE culture IS NULL;';
+      QubitPdo::modify(
+        sprintf($sql, $i18nTable),
+        array(':cul' => $defaultCulture)
```

Comment:
I would suggest setting the `source_culture` first (e.g. if source_culture is NULL, set it to default culture) *then* setting the `i18n.culture = source_culture`.   In most cases if there is no culture set, I would assume the culture should be the same as the row source_culture, rather than the default culture.  

Practically, I'm sure in most cases when the culture == null the source_culture == null too but I'm not sure if we can assume it.

Reply:
Good catch, thanks @djjuhasz!